File: c++/web_tutorial_mastercopy/eq_test.cpp

#include <string>

using namespace std;

#include "Bridges.h"
#include "DataSource.h"
#include "data_src/EarthquakeUSGS.h"


using namespace bridges;


// This program fragment illustrates how to access the USGS earthquake data
int main(int argc, char **argv) {

	// create Bridges object
#if TESTING
	// command line args provide credentials and server to test on
	Bridges bridges (atoi(argv[1]), argv[2], argv[3]);
	if (argc > 4)
		bridges.setServer(argv[4]);
#else
	Bridges bridges (YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID",
		"YOUR_API_KEY");
#endif
	// set title
	bridges.setTitle("Accessing USGIS Earthquake Data (USGIS Data)");

	// read the earth quake  data
	DataSource ds (&bridges);
	int max_quakes = 100;
	vector<EarthquakeUSGS> eq_list = ds.getEarthquakeUSGSData(max_quakes);

	// print the first quake record
	for (int k = 0; k < 3; k++) {
		cout << "Earthquake " << k << ": \n";
		cout << "\tMagnitude:" << eq_list[0].getMagnitude() << endl
		<< "\tDate:" << eq_list[0].getDateStr() << endl
		<< "\tLocation: " <<  eq_list[0].getLocation() << endl
		<< "\tLat/Long:"  << eq_list[0].getLatit() << "," <<
		eq_list[0].getLongit() << endl;
	}

	eq_list = ds.getEarthquakeUSGSData(); //This is a different code path

	// print the first quake record
	for (int k = 0; k < 3; k++) {
		cout << "Earthquake " << k << ": \n";
		cout << "\tMagnitude:" << eq_list[0].getMagnitude() << endl
		<< "\tDate:" << eq_list[0].getDateStr() << endl
		<< "\tLocation: " <<  eq_list[0].getLocation() << endl
		<< "\tLat/Long:"  << eq_list[0].getLatit() << "," <<
		eq_list[0].getLongit() << endl;
	}

	

	return 0;
}

File: java/web_tutorial_mastercopy/eq_test.java
import java.util.List;
import bridges.connect.Bridges;
import bridges.connect.DataSource;
import bridges.data_src_dependent.EarthquakeUSGS;

// This program fragment illustrates how to access the USGS earthquake data
public class eq_test {
	public static void main(String[] args) throws Exception {

		// create Bridges object
#if TESTING
		// command line args provide credentials and server type
		Bridges bridges = new Bridges(Integer.parseInt(args[0]), args[1], args[2]);
		bridges.setServer(args[3]);
#else
		Bridges bridges = new Bridges (YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY");
#endif
		// read the earth quake  data and build the BST
		bridges.setTitle("Accessing USGIS Earthquake Data (USGS Data)");

		DataSource ds = bridges.getDataSource();

		// get 10 most recent earthquake records
		List<EarthquakeUSGS> eq_list = ds.getEarthquakeUSGSData(10);

		// print the first record
		System.out.println("Earthquake 0:");
		System.out.println ("\tMagnitude:"  + eq_list.get(0).getMagnitude() + "\n"
			+ "\tDate:" + eq_list.get(0).getTime() + "\n"
			+ "\tLocation: " +  eq_list.get(0).getLocation() + "\n"
			+ "\tLat/Long:"  + eq_list.get(0).getLatit() + "," +
			eq_list.get(0).getLongit());


		//get all earthquakes
		eq_list = ds.getEarthquakeUSGSData();

		// print the first record
		System.out.println("Earthquake 0:");
		System.out.println ("\tMagnitude:"  + eq_list.get(0).getMagnitude() + "\n"
			+ "\tDate:" + eq_list.get(0).getTime() + "\n"
			+ "\tLocation: " +  eq_list.get(0).getLocation() + "\n"
			+ "\tLat/Long:"  + eq_list.get(0).getLatit() + "," +
			eq_list.get(0).getLongit());
	}
}
